### ISSUE TITLE
Support for creating child windows

### ISSUE THREAD

**Issue by Boscop** [gabdube/native-windows-gui#33]
It would be very useful to have support for creating child windows with a given parent handle.
I implemented support for child windows on Windows in my glutin fork like half a year ago:
https://github.com/Boscop/glutin/blob/master/src/window.rs#L205
https://github.com/Boscop/glutin/blob/master/src/api/win32/init.rs#L76-L84
https://github.com/Boscop/glutin/blob/master/src/api/win32/callback.rs#L23
https://github.com/Boscop/glutin/blob/master/src/api/win32/mod.rs#L471-L473
(Child windows have to run in the same thread as the parent window, so I'm using a HashMap with HWND as key for event dispatching of all the windows in the same thread.)

It would also be useful to have support for timers on the window via a callback:
https://github.com/Boscop/glutin/blob/master/src/window.rs#L508-L516

I implemented this because it's necessary to create a child window for writing GUIs for VST plugins. The plugin host creates a window and [passes it](https://github.com/overdrivenpotato/rust-vst2/blob/master/src/editor.rs#L23) to the plugin, which has to create a child window on that for it's GUI and register a timer on the HWND so that it refreshes it's GUI every few milliseconds.
This is what I do in the plugin:
```rust
impl<'a> Editor for MyPlugin<'a> {
	fn open(&mut self, parent: *mut c_void) {
		let mut display = WindowBuilder::new().with_dimensions(WINDOW_WIDTH, WINDOW_HEIGHT).with_parent(WindowID::new(parent)).build_glium().unwrap();
		{
			let window = display.get_window().unwrap();
			unsafe { user32::SetWindowLongA(window.platform_window() as winapi::HWND, winapi::winuser::GWLP_USERDATA, self as *mut _ as winapi::LONG); }
			window.set_timer(WIN_GUI_FRAME_TIMER, 25 /* ms */, Some(handle_timer)).unwrap();
		}
// ...
const WIN_GUI_FRAME_TIMER: winapi::UINT_PTR = 4242;
unsafe extern "system" fn handle_timer(hwnd: winapi::HWND, _: winapi::UINT, timer_id: winapi::UINT_PTR, elapsed_ms: winapi::DWORD) {
	if timer_id != WIN_GUI_FRAME_TIMER {
		return;
	}
	let plugin = &mut *(user32::GetWindowLongA(hwnd, winapi::winuser::GWLP_USERDATA) as *mut MyPlugin);
	plugin.render_ui_frame();
}
```
It would be very useful to have support for creating child windows in NWG which can be rendered one frame at a time.
So that NWG can be used to write GUIs for VST plugins, completely in Rust :)

**Comment by gabdube** [comment]
By `creating a child window`, are you thinking, for example, to add an optional `parent` field to the window template? Or to add a new window template that only accepts a HWND parameter? Or both?

It's not the first issue that mention wrapping external Window, so I'm seriously considering implementing it (and it wouldn't even be hard).

For the timer, I'm not sure I understand the whole concept. Currently the NWG timers are bound to the UI (hidden) inner window. In order for a VST to work correcty, the programmer needs be able to select window bound to the timer?

**Comment by Boscop** [comment]
In glutin I added a method like `fn with_parent(mut self, parent: super::WindowID) -> WindowBuilder`.
I'm not sure what the most ergonomic way would be in NWG. A builder only makes sense if there are multiple optional options. The use case with child windows is rare, so it would probably warrant introducing a different function for it, so that in 90% of cases people can call the normal function instead of passing `None` to it.

For timers, it's enough to expose the HWND to the client code, so that the client code can pass it to the winapi functions to register the timer etc.
In glutin I added the `set_timer()` function for convenience:
https://github.com/Boscop/glutin/blob/master/src/api/win32/mod.rs#L372
